### PR TITLE
Added documentation of how to handle code changes.

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
This is an initial suggestion that should be reviewed.

When agreed, the suggested API should be implemented.

## Type of change
- [ ] Bug fix & code cleanup
- [ ] New feature
- [x] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: doc/contributors_guide/code_changes.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    find \( -name '*.py' -o -name '*.i' -o -name '*.c' \) ! -path './build*' | xargs grep -n -A1 deprecation_warning
+
+
+How to handle different types of code changes
+---------------------------------------------
+
+### Backward-compatible API changes
+The easiest API changes are those that only adds new functions or classes without changing the existing code, apart from adding deprecation warnings hinting about how to use the new API.
+
+Adding new arguments to an existing function or new methods to existing classes falls also into this category.  It is possible to
```
Comment (torhaugl) [inline doc/contributors_guide/code_changes.md:39]:
```suggestion
Adding new arguments to an existing function or new methods to existing classes falls also into this category.
```

File: doc/contributors_guide/code_changes.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+----------
+Before reaching v 1.0.0, DLite follows the following versioning rules that are well-known from Python:
+
+* **patch release**: Only backward-compatible changes.
+  New deprecation warnings can be added.
+
+* **minor release**: Backward-incompatible changes.
+  Removal of deprecated features.
+
+After version 1.0.0, DLite should strictly follow the [semantic versioning] rules.
```
Comment (torhaugl) [inline doc/contributors_guide/code_changes.md:16]:
```suggestion
After version 1.0.0, DLite should strictly follow the [semantic versioning](https://semver.org/) rules.
```

File: doc/contributors_guide/code_changes.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+- **via environment variables**:
+
+      export DLITE_BEHAVIOR_<NAME>
+
+A deprecation warning will automatically be issued if a behavior is not selected
+explicitly.
+
+
+[semantic versioning]: https://semver.org/
```
Comment (torhaugl) [inline doc/contributors_guide/code_changes.md:75]:
```suggestion
```

File: doc/contributors_guide/code_changes.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Versioning
+----------
+Before reaching v 1.0.0, DLite follows the following versioning rules that are well-known from Python:
+
+* **patch release**: Only backward-compatible changes.
+  New deprecation warnings can be added.
+
+* **minor release**: Backward-incompatible changes.
+  Removal of deprecated features.
```
Comment (torhaugl) [inline doc/contributors_guide/code_changes.md:14]:
```suggestion
  Removal of deprecated features.
  Addition of new features.
```

**Review by torhaugl — COMMENTED** [review]
Added some suggestions

File: doc/contributors_guide/code_changes.md (hunk 2 above)
Comment (jesper-friis) [reply doc/contributors_guide/code_changes.md:17]: Personlig synes jeg at teksten blir mer leselig når url-referansene legges i slutten av dokumentet.

File: doc/contributors_guide/code_changes.md (hunk 3 above)
Comment (jesper-friis) [reply doc/contributors_guide/code_changes.md:76]: Se kommentaren over

**Pull request merged**